app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,346 @@
+ * Error UI to show if there is one of the errors outlined in [SyncedTabsView.ErrorType].
+ *
+ * @param errorType [SyncedTabsView.ErrorType] to be displayed after mapping to a string resource.
+ * @param navController [NavController] to call [NavGraphDirections.actionGlobalTurnOnSync()].
+ */
+@Composable
+fun SyncedTabsErrorItem(errorType: SyncedTabsView.ErrorType, navController: NavController? = null) {
+    val width: Float
+    val dashLength: Float
+    with(LocalDensity.current) {
+        width = 2.dp.toPx()
+        dashLength = 4.dp.toPx()
+    }
+    // As of December 2021, there is not an official way to get a rounded rectangle border with
+    // dashes using Compose. This is a next-best attempt with hard corners.
+    val stroke = Stroke(
+        width = width,
+        pathEffect = PathEffect.dashPathEffect(floatArrayOf(dashLength, dashLength), 0f)
+    )
+    val dashColor = FirefoxTheme.colors.borderDivider
+    Box(
+        Modifier
+            .padding(20.dp)
+            .height(IntrinsicSize.Min)
+    ) {
+        Canvas(Modifier.fillMaxSize()) {
+            drawRoundRect(color = dashColor, style = stroke)
+        }
+        Column(
+            modifier = Modifier
+                .padding(16.dp)
+                .fillMaxWidth()
+        ) {
+            PrimaryText(
+                text = stringResource(id = errorType.toStringRes()),
+                fontSize = 14.sp,
+                modifier = Modifier
+                    .padding(0.dp, 4.dp, 0.dp, 0.dp)
+                    .fillMaxWidth()
+            )
+            if (navController != null) {
+                Spacer(modifier = Modifier.height(8.dp))
+                Row(
+                    modifier = Modifier
+                        .clip(RoundedCornerShape(4.dp))
+                        .background(FirefoxTheme.colors.actionPrimary)
+                        .padding(8.dp, 4.dp)
+                        .clickable(
+                            onClickLabel = stringResource(id = R.string.synced_tabs_sign_in_button),
+                            onClick = {
+                                navController.navigate(NavGraphDirections.actionGlobalTurnOnSync())
+                            }
+                        )
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_sign_in),
+                        contentDescription = null,
+                        tint = FirefoxTheme.colors.textInverted,
+                    )
+                    Spacer(modifier = Modifier.width(8.dp))
+                    Text(
+                        text = stringResource(id = R.string.synced_tabs_sign_in_button),
+                        color = FirefoxTheme.colors.textInverted,
+                        fontSize = 14.sp,
+                        fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+                        maxLines = 2,
+                        modifier = Modifier.align(Alignment.CenterVertically)
```

Comment:
Can we reorder the `modifier` to be below `text`? I usually just follow the parameter ordering that is defined in `Text`, but otherwise, I think you went with alphabetical ordering except you prioritized `text` as the first parameter since that's the most important parameter. Only calling this out because I want to make sure we are somewhat consistent and not investing a new ordering of listing out compose parameters that is not reproducible by others.